src/benchmarks/micro/libraries/System.Text.RegularExpressions/Perf.Regex.Industry.cs

```diff
@@ -0,0 +1,302 @@
+#endif
+            )]
+        public RegexOptions Options { get; set; }
+
+        private string _3200;
+        private Regex _regex;
+
+        [GlobalSetup(Target = nameof(Count))]
+        public void Setup()
+        {
+            _regex = new Regex(Pattern, Options);
+            _3200 = Perf_Regex_Industry.ReadInputFile("3200.txt.gz");
+        }
+
+        [Benchmark]
+        public int Count() => Perf_Regex_Industry.Count(_regex, _3200);
+    }
+
+    /// <summary>Performance tests adapted from https://www.boost.org/doc/libs/1_41_0/libs/regex/doc/gcc-performance.html</summary>
+    [BenchmarkCategory(Categories.Libraries)]
+    [SkipTooManyTestCasesValidator]
+    public class Perf_Regex_Industry_BoostDocs_Simple
+    {
+        [Params(0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13)]
+        public int Id { get; set; }
+
+        [Params(
+            RegexOptions.None,
+            RegexOptions.Compiled
+#if NET7_0_OR_GREATER
+            , RegexOptions.NonBacktracking
+#endif
+            )]
+        public RegexOptions Options { get; set; }
+
+        private string _input;
+        private Regex _regex;
+
+        [GlobalSetup(Target = nameof(IsMatch))]
+        public void Setup()
+        {
+            _input = Id switch
+            {
+                0 => @"abc",
+                1 => @"100- this is a line of ftp response which contains a message string",
+                2 => @"1234-5678-1234-456",
+                3 => @"john@johnmaddock.co.uk",
+                4 => @"foo12@foo.edu",
+                5 => @"bob.smith@foo.tv",
+                6 => @"EH10 2QQ",
+                7 => @"G1 1AA",
+                8 => @"SW1 1ZZ",
+                9 => @"4/1/2001",
+                10 => @"12/12/2001",
+                11 => @"123",
+                12 => @"+3.14159",
+                _ => @"-3.14159",
+            };
+
+            _regex = new Regex(Id switch
+            {
+                0  => @"abc",
+                1  => @"^([0-9]+)(\-| |$)(.*)$",
+                2  => @"(\d{4}[- ]){3}\d{3,4}",
+                3  => @"^([a-zA-Z0-9_\-\.]+)@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.)|(([a-zA-Z0-9\-]+\.)+))([a-zA-Z]{2,4}|[0-9]{1,3})(\]?)$",
+                4  => @"^([a-zA-Z0-9_\-\.]+)@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.)|(([a-zA-Z0-9\-]+\.)+))([a-zA-Z]{2,4}|[0-9]{1,3})(\]?)$",
+                5  => @"^([a-zA-Z0-9_\-\.]+)@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.)|(([a-zA-Z0-9\-]+\.)+))([a-zA-Z]{2,4}|[0-9]{1,3})(\]?)$",
+                6  => @"^[a-zA-Z]{1,2}[0-9][0-9A-Za-z]{0,1} {0,1}[0-9][A-Za-z]{2}$",
+                7  => @"^[a-zA-Z]{1,2}[0-9][0-9A-Za-z]{0,1} {0,1}[0-9][A-Za-z]{2}$",
+                8  => @"^[a-zA-Z]{1,2}[0-9][0-9A-Za-z]{0,1} {0,1}[0-9][A-Za-z]{2}$",
+                9  => @"^\d{1,2}/\d{1,2}/\d{4}$",
+                10 => @"^\d{1,2}/\d{1,2}/\d{4}$",
+                11 => @"^[-+]?\d*\.?\d*$",
+                12 => @"^[-+]?\d*\.?\d*$",
+                _  => @"^[-+]?\d*\.?\d*$",
```

Comment:
Can we use the C# 9.0 syntax and remove the duplicaiton?

```suggestion
                3 or 4 or 5 => @"^([a-zA-Z0-9_\-\.]+)@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.)|(([a-zA-Z0-9\-]+\.)+))([a-zA-Z]{2,4}|[0-9]{1,3})(\]?)$",
                6 or 7 or 8 => @"^[a-zA-Z]{1,2}[0-9][0-9A-Za-z]{0,1} {0,1}[0-9][A-Za-z]{2}$",
                9 or 10 => @"^\d{1,2}/\d{1,2}/\d{4}$",
                11 or 12 or _ => @"^[-+]?\d*\.?\d*$",
```